### PR TITLE
More UX Tweak

### PR DESCRIPTION
**Pull request by youwenliang**
1. Handle mic permission blocked error case
2. Limit the number of results shown in the dropdown list
3. Add some transitions to the mic character
4. UI tweaks and voice visualization tweaks
5. Input field UX tweaks

### CODE REVIEW

**Comment by andrenatal** [comment]
Hey @johngruen. Can you please review and merge this one as well?

**Comment by johngruen** [comment]
@youwenliang this looks great with a few caveats:
1. Gotta rebase!
2. The feedback button is real twitchy on the edges
twitch
3. The mic animation is a little twitchy also (i don't think this blocks)

also some comments inline. Clean this up and I'll merge tomorrow morning.

File: extension/content.js [hunk 1]
```diff
@@ -177,10 +179,29 @@
 
                 input.focus();
 
+                input.addEventListener("keypress", function _expand_input(e) {
+                    // e.preventDefault();
+                    if(e.keyCode == 13) {
```
Comment (johngruen) [inline extension/content.js]: Should these be strict equality `===`?

File: extension/content.js [hunk 2]
```diff
@@ -177,10 +179,29 @@
...
+                        e.preventDefault();
+                        list.classList.add('close');
+                        resolve(input.value);
+                    }
+                    else if(e.keyCode == 8) {
```
Comment (johngruen) [inline extension/content.js]: `===`

File: extension/content.js [hunk 3]
```diff
@@ -177,10 +179,29 @@
...
+                    }
+                    else if(e.keyCode < 37 || e.keyCode > 40) {
+                        input.size = input.value.length;
+                        list.style.width = `${input.offsetWidth}px`;
+                    }
+                });
+
+
                 form.addEventListener("submit", function _submit_form(e) {
                     console.log('!!!!!!!!');
```
Comment (johngruen) [inline extension/content.js]: I think this is my fault, but can you knock it out?

**Comment by youwenliang** [comment]
@johngruen Check if this is able to merge. I'm not very familiar with rebase so hopefully this didn't erase someone else's work...

**Comment by johngruen** [comment]
@youwenliang this looks like it's working, i'm not sure why there are 18 commits on this branch :)

**Comment by johngruen** [comment]
@youwenliang clicking the close and reset buttons do not work for me on this branch unfortunately.

**Comment by youwenliang** [comment]
@johngruen I'll take a look!

**Comment by johngruen** [comment]
@youwenliang the submit button does not submit

**Comment by johngruen** [comment]
@youwenliang i know this is a pain, but i would probably recommend starting this over on a clean branch. Your PR is somehow all mixed up with other code (not sure why). Easiest thing might be to start over from latest master and change on the correct parts.

**Comment by youwenliang** [comment]
@johngruen I can do that. But can you check if the current master branch works normally? This branch is rebased from the current master.

**Comment by johngruen** [comment]
@youwenliang master works fine

**Comment by johngruen** [comment]
@youwenliang huh, actually does not seem to work all the time :(

**Pull request closed without merging**